lib/compat/wordpress-6.5/class-wp-navigation-block-renderer.php

```diff
@@ -0,0 +1,611 @@
+		if ( ! wp_script_is( $view_js_file ) ) {
+			$script_handles = $block->block_type->view_script_handles;
+
+			// If the script is not needed, and it is still in the `view_script_handles`, remove it.
+			if ( ! $should_load_view_script && in_array( $view_js_file, $script_handles, true ) ) {
+				$block->block_type->view_script_handles = array_diff( $script_handles, array( $view_js_file ) );
+			}
+			// If the script is needed, but it was previously removed, add it again.
+			if ( $should_load_view_script && ! in_array( $view_js_file, $script_handles, true ) ) {
+				$block->block_type->view_script_handles = array_merge( $script_handles, array( $view_js_file ) );
+			}
+		}
+	}
+
+	/**
+	 * Returns the markup for the navigation block.
+	 *
+	 * @param array         $attributes The block attributes.
+	 * @param WP_Block_List $inner_blocks The list of inner blocks.
+	 * @return string Returns the navigation wrapper markup.
+	 */
+	private static function get_wrapper_markup( $attributes, $inner_blocks ) {
+		$inner_blocks_html = static::get_inner_blocks_html( $attributes, $inner_blocks );
+		if ( static::is_responsive( $attributes ) ) {
+			return static::get_responsive_container_markup( $attributes, $inner_blocks, $inner_blocks_html );
+		}
+		return $inner_blocks_html;
+	}
+
+	/**
+	 * Renders the navigation block.
+	 *
+	 * @param array    $attributes The block attributes.
+	 * @param string   $content    The saved content.
+	 * @param WP_Block $block      The parsed block.
+	 * @return string Returns the navigation block markup.
+	 */
+	public static function render( $attributes, $content, $block ) {
+		static $seen_menu_names = array();
+
+		$nav_menu_name = static::get_navigation_name( $attributes, $seen_menu_names );
+
+		/**
+		 * Deprecated:
+		 * The rgbTextColor and rgbBackgroundColor attributes
+		 * have been deprecated in favor of
+		 * customTextColor and customBackgroundColor ones.
+		 * Move the values from old attrs to the new ones.
+		 */
+		if ( isset( $attributes['rgbTextColor'] ) && empty( $attributes['textColor'] ) ) {
+			$attributes['customTextColor'] = $attributes['rgbTextColor'];
+		}
+
+		if ( isset( $attributes['rgbBackgroundColor'] ) && empty( $attributes['backgroundColor'] ) ) {
+			$attributes['customBackgroundColor'] = $attributes['rgbBackgroundColor'];
+		}
+
+		unset( $attributes['rgbTextColor'], $attributes['rgbBackgroundColor'] );
+
+		$inner_blocks = static::get_inner_blocks( $attributes, $block );
+		// Prevent navigation blocks referencing themselves from rendering.
+		if ( block_core_navigation_block_contains_core_navigation( $inner_blocks ) ) {
+			return '';
+		}
+
+		// If the menu name has been used previously then append an ID
+		// to the name to ensure uniqueness across a given post.
+		if ( isset( $seen_menu_names[ $nav_menu_name ] ) && $seen_menu_names[ $nav_menu_name ] > 1 ) {
+			$count         = $seen_menu_names[ $nav_menu_name ];
+			$nav_menu_name = $nav_menu_name . ' ' . ( $count );
+		}
```

Comment:
Method here? `get_unique_navigation_name`?

Reply:
Done in b16a3bd552